examples/nanocoap_server/coap_handler.c

```diff
@@ -6,20 +6,48 @@
  * directory for more details.
  */
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 
 #include "nanocoap.h"
 
+/* internal value that can be read/written via CoAP */
+static uint8_t internal_value = 0;
+
 static ssize_t _riot_board_handler(coap_pkt_t *pkt, uint8_t *buf, size_t len)
 {
     return coap_reply_simple(pkt, buf, len,
             COAP_FORMAT_TEXT, (uint8_t*)RIOT_BOARD, strlen(RIOT_BOARD));
 }
 
+static ssize_t _riot_value_handler(coap_pkt_t *pkt, uint8_t *buf, size_t len)
+{
+    ssize_t p = 0;
+    char rsp[64];
+    p += sprintf(rsp, "value: %d", internal_value);
+    rsp[p] = '\0';
+
+    return coap_reply_simple(pkt, buf, len,
+            COAP_FORMAT_TEXT, (uint8_t*)rsp, strlen(rsp));
+}
+
+static ssize_t _riot_value_put_handler(coap_pkt_t *pkt, uint8_t *buf, size_t len)
+{
+    char msg[14] = "Value updated\0";
+
+    /* convert the payload to an integer */
+    internal_value = strtol((char*)pkt->payload, NULL, 10);
+
+    return coap_reply_simple(pkt, buf, len,
+            COAP_FORMAT_TEXT, (uint8_t*)msg, strlen(msg));
+}
+
 const coap_resource_t coap_resources[] = {
     COAP_WELL_KNOWN_CORE_DEFAULT_HANDLER,
     { "/riot/board", COAP_GET, _riot_board_handler },
+    { "/riot/value", COAP_GET, _riot_value_handler },
```

Comment:
> Does this work with two entries for the same path?

yes it does

> Anyways, the method field is a bitfield. If it makes things easier, you could use only one handler, mark it as "COAP_GET | COAP_PUT", and then switch() over the method within that handler.

I pushed an update addressing this comment. Surprisingly the issue with payload not being reset is gone with this change. Weird.